Sources/ATTNAPI.m

```diff
@@ -42,6 +95,95 @@ - (void)sendUserIdentity:(ATTNUserIdentity *)userIdentity domain:(NSString *)dom
     }];
 }
 
+- (void)sendEvent:(id<ATTNEvent>)event userIdentity:(ATTNUserIdentity*)userIdentity domain:(NSString*) domain {
+    [self getGeoAdjustedDomain:domain completionHandler:^(NSString* geoAdjustedDomain, NSError* error) {
+        if (error) {
+            NSLog(@"Error sending user identity: '%@'.", error);
+            
+        }
+        
+        [self sendEventInternal:event userIdentity:userIdentity domain:domain];
+    }];
+}
+
+- (void)sendEventInternal:(id<ATTNEvent>)event userIdentity:(ATTNUserIdentity*)userIdentity domain:(NSString*) domain {
+    // slice up the Event into individual EventRequests
+    NSArray<EventRequest *>* requests = [self convertEventToRequests:event];
+    
+    for (EventRequest* request in requests) {
+        [self sendEventInternalForRequest:request userIdentity:userIdentity domain:domain];
```

Comment:
does each one of these requests open the radio? should we be working on reducing the number of requests rn or you think it's fine?

Reply:
Honestly I think events are so infrequent right now that it won't matter. The radio should stay on between requests since they're back to back, and the radio may even still be on from the app's other communications with their own servers.